.github/workflows/ci.yml

```diff
@@ -0,0 +1,44 @@
+name: Continuous Integration
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the main branch
+  # See: https://github.community/t/duplicate-checks-on-push-and-pull-request-simultaneous-event/18012
+  push:
+    branches:
+      - main*
+  pull_request:
+    branches:
+      - main*
+
+jobs:
+  build:
+    name: Full Build (node ${{ matrix.node_version }})
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node_version: [16.x, 18.x]
+      # https://stackoverflow.com/questions/61070925/github-actions-disable-auto-cancel-when-job-fails
+      fail-fast: false
+
+    steps:
+      # using `v1` because of: https://github.com/actions/checkout/issues/246
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node_version }}
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Test
+        run: npm run test
+
+      - name: Coveralls
+        uses: coverallsapp/github-action@master
```

Comment:
should we use a specific version instead of `@master`

Reply:
done

Reply:
done